### ISSUE TITLE
CORS header 'Access-Control-Allow-Origin' missing

### ISSUE THREAD

**Issue by devopsader** [FortAwesome/Font-Awesome#17402]
I'm suddenly getting errors on my kit, i tried the alternative using CDN, and that also did not work.

I'm getting the following response on page load:

```
XHROPTIONShttps://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css
[HTTP/2 403 Forbidden 1094ms]
OPTIONS
	https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css
Status403
Forbidden
VersionHTTP/2
Transferred0.98 kB (0 B size)
    HTTP/2 403 Forbidden
    date: Fri, 13 Nov 2020 14:35:01 GMT
    content-type: text/html; charset=UTF-8
```

When trying the `example.html` file:

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css. (Reason: CORS header 'Access-Control-Allow-Origin' missing).
```

**Comment by leolandotan** [comment]
Im having the same issue as well.

**Comment by yan3321** [comment]
Currently facing the same problem. Thought something was up with CORS on my server and did some fiddling, but issues still persist.

**Comment by tagliala** [comment]
I've reported this issue to hello@fontawesome.com, let's wait for feedback

**Comment by robmadole** [comment]
Can you try and run the following command and provide the output to me?

```
curl 'https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css' \
  -X 'OPTIONS' \
  -H 'authority: ka-f.fontawesome.com' \
  -H 'accept: */*' \
  -H 'access-control-request-method: GET' \
  -H 'origin: http://localhost' \
  -H 'referer: http://localhost' \
  --compressed -v
```

**Comment by yan3321** [comment]
> Can you try and run the following command and provide the output to me?
> 
> ```
> curl 'https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css' \
>   -X 'OPTIONS' \
>   -H 'authority: ka-f.fontawesome.com' \
>   -H 'accept: */*' \
>   -H 'access-control-request-method: GET' \
>   -H 'origin: http://localhost' \
>   -H 'referer: http://localhost' \
>   --compressed -v
> ```

Here's my output from the given script:
```
*   Trying 2606:4700:e6::ac40:ca1c:443...
* TCP_NODELAY set
* Connected to ka-f.fontawesome.com (2606:4700:e6::ac40:ca1c) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=CA; L=San Francisco; O=Cloudflare, Inc.; CN=sni.cloudflaressl.com
*  start date: Oct 13 00:00:00 2020 GMT
*  expire date: Oct 12 23:59:59 2021 GMT
*  subjectAltName: host "ka-f.fontawesome.com" matched cert's "ka-f.fontawesome.com"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x7fffbd16baa0)
> OPTIONS /releases/v5.15.1/css/free.min.css HTTP/2
> Host: ka-f.fontawesome.com
> user-agent: curl/7.68.0
> accept-encoding: deflate, gzip, br
> authority: ka-f.fontawesome.com
> accept: */*
> access-control-request-method: GET
> origin: http://localhost
> referer: http://localhost
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 200
< date: Wed, 18 Nov 2020 17:12:00 GMT
< content-length: 0
< set-cookie: __cfduid=dabf8d2cf94dd59ab2f80c132b8aad38b1605719520; expires=Fri, 18-Dec-20 17:12:00 GMT; path=/; domain=.ka-f.fontawesome.com; HttpOnly; SameSite=Lax
< access-control-allow-origin: *
< access-control-allow-methods: GET
< access-control-allow-headers: fa-kit-token
< access-control-max-age: 3000
< x-cache: Hit from cloudfront
< via: 1.1 46ddd60a6cc4c84ac6b72be6dae3d48b.cloudfront.net (CloudFront)
< x-amz-cf-pop: HKG62-C2
< x-amz-cf-id: P8x6GeDqdz1c6PY-L3956uMSdIuIoWf1LQEWjHBOTHAg21f9LRtI7w==
< age: 2561
< cf-cache-status: DYNAMIC
< cf-request-id: 067df08e07000021a651ab8000000001
< expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=M9MPT8UZNbj9wJnnb%2Fk9ZrybN69xz4TQJlzDDxhCkQbdssFPND1p7X7A3NlX0HeBpGX6Rbbhk%2FEv%2B5QL1RogPyiKYh0IFk6chlk3mQGaj9wiC7JraFl48X%2B8IfBYo03ohA%3D%3D"}],"group":"cf-nel","max_age":604800}
< nel: {"report_to":"cf-nel","max_age":604800}
< server: cloudflare
< cf-ray: 5f43505cdce021a6-HKG
<
* Connection #0 to host ka-f.fontawesome.com left intact
```

**Comment by yan3321** [comment]
If it helps, here's the (likely) relevant response and request headers:
![image](https://user-images.githubusercontent.com/9079480/99566447-e16bad00-2a07-11eb-9ac4-e0234e78e7d8.png)

Response:
```
HTTP/2 200 OK
date: Wed, 18 Nov 2020 17:23:03 GMT
content-type: text/css
set-cookie: __cfduid=d400d1717efd2c80489ec2eb4c2736b1e1605720183; expires=Fri, 18-Dec-20 17:23:03 GMT; path=/; domain=.ka-f.fontawesome.com; HttpOnly; SameSite=Lax
last-modified: Wed, 14 Oct 2020 21:18:07 GMT
etag: W/"319d424ba89a84bbd230a3b5f7024193"
cache-control: max-age=31556926
vary: Accept-Encoding
x-cache: Hit from cloudfront
via: 1.1 563e25ff4039f663425caf2d104a488e.cloudfront.net (CloudFront)
x-amz-cf-pop: HKG62-C2
x-amz-cf-id: 8zgUgQZV5CoN_05j3KN_maxd-Jd6xZmbqD1E3GMWW3D8z8Dawn_DHQ==
age: 32529
cf-cache-status: HIT
cf-request-id: 067dfaab7c0000196933a44000000001
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=cIjAWN6yDkDprOTKIYFsffJUVb1bBgdhyr7Gq%2FG%2BI1Z7Nx%2BVb3UrZ7cIY39aSDpC3bxSo0jSP0nGYrQ2lT38twTEtt%2B35Lpp4jvSHx0lBqb2FxsxqbCowxp4PsJXt8Gm6g%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 5f43608bfe0d1969-HKG
content-encoding: br
X-Firefox-Spdy: h2
```
Looks like it's missing the CORS header there.

Request:
```
GET /releases/v5.15.1/css/free.min.css HTTP/2
Host: ka-f.fontawesome.com
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:84.0) Gecko/20100101 Firefox/84.0
Accept: */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Referer: https://demo.libramex.tech/login_form/login_form.html
fa-kit-token: a81368914c
Origin: https://demo.libramex.tech
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
TE: Trailers
```

**Comment by robmadole** [comment]
@yan3321 can you post the output of this?

```
curl 'https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css' \
  -X 'GET' \
  -H 'accept: */*' \
  -H 'origin: https://demo.libramex.tech'
```

**Comment by yan3321** [comment]
> curl 'https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css' \
>   -X 'GET' \
>   -H 'accept: */*' \
>   -H 'origin: https://demo.libramex.tech'

@robmadole Output gave me the CSS contents below (truncated since too large):

```css
/*!
 * Font Awesome Free 5.15.1 by @fontawesome - https://fontawesome.com
 * License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
 */.fa,.fab,.fad,.fal,.far,.fas{-moz-osx-font-smoothing:grayscale;-webkit-font-smoothing:antialiased;display:inline-block;font-style:normal;font-variant:normal;text-rendering:auto;line-height:1}.fa-lg{font-size:1.33333em;line-height:.75em;vertical-align:-.0667em}.fa-xs{font-size:.75em}.fa-sm{font-size:.875em}.fa-1x{font-size:1em}.fa-2x{font-size:2em}.fa-3x{font-size:3em}.fa-4x{font-size:4em}.fa-5x{font-size:5em}.fa-6x{font-size:6em}.fa-7x{font-size:7em}.fa-8x{font-size:8em}.fa-9x{font-size:9em}.fa-10x{font-size:10em}.fa-fw{text-align:center;width:1.25em}.fa-ul{list-style-type:none;margin-left:2.5em;padding-left:0}.fa-ul>li{position:relative}.fa-li{left:-2em;position:absolute;text-align:center;width:2em;line-height:inherit}.fa-border{border:.08em solid #eee;border-radius:.1em;padding:.2em .25em .15em}.fa-pull-left{float:left}.fa-pull-right{float:right}.fa.fa-pull-left,.fab.fa-pull-left,.fal.fa-pull-left,.far.fa-pull-left,.fas.fa-pull-left{margin-right:.3em}.fa.fa-pull-right,.fab.fa-pull-right,.fal.fa-pull-right,.far.fa-pull-right,.fas.fa-pull-right{margin-left:.3em}...
```

Not sure if that helps though, so I also tried appending `--compressed -v` to the curl, running this:
```
curl 'https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css' \
  -X 'GET' \
  -H 'accept: */*' \
  -H 'origin: https://demo.libramex.tech' \
  --compressed -v
```
Additional output above the CSS contents:
```
*   Trying 2606:4700:e6::ac40:ca1c:443...
* TCP_NODELAY set
* Connected to ka-f.fontawesome.com (2606:4700:e6::ac40:ca1c) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=CA; L=San Francisco; O=Cloudflare, Inc.; CN=sni.cloudflaressl.com
*  start date: Oct 13 00:00:00 2020 GMT
*  expire date: Oct 12 23:59:59 2021 GMT
*  subjectAltName: host "ka-f.fontawesome.com" matched cert's "ka-f.fontawesome.com"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x7fffed593aa0)
> GET /releases/v5.15.1/css/free.min.css HTTP/2
> Host: ka-f.fontawesome.com
> user-agent: curl/7.68.0
> accept-encoding: deflate, gzip, br
> accept: */*
> origin: https://demo.libramex.tech
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 200
< date: Wed, 18 Nov 2020 18:37:02 GMT
< content-type: text/css
< set-cookie: __cfduid=d6c25ce6e5651bbf89619c3a14dbb70be1605724622; expires=Fri, 18-Dec-20 18:37:02 GMT; path=/; domain=.ka-f.fontawesome.com; HttpOnly; SameSite=Lax
< last-modified: Wed, 14 Oct 2020 21:18:07 GMT
< etag: W/"319d424ba89a84bbd230a3b5f7024193"
< cache-control: max-age=31556926
< vary: Accept-Encoding
< x-cache: Hit from cloudfront
< via: 1.1 563e25ff4039f663425caf2d104a488e.cloudfront.net (CloudFront)
< x-amz-cf-pop: HKG62-C2
< x-amz-cf-id: 8zgUgQZV5CoN_05j3KN_maxd-Jd6xZmbqD1E3GMWW3D8z8Dawn_DHQ==
< age: 36968
< cf-cache-status: HIT
< cf-request-id: 067e3e64c8000024288e909000000001
< expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=%2FUinLW8XT1J07ajz8HFEwcMu7XmE87MMMEXsZtFTS2qaeHHKVDgAPwZuw%2FDV4lqJBLKd6TZHb2U%2FWOOnJLuOG71v0out3UwlGRhadZqks4JySgzVvMmS605nn60HXJkosg%3D%3D"}],"group":"cf-nel","max_age":604800}
< nel: {"report_to":"cf-nel","max_age":604800}
< server: cloudflare
< cf-ray: 5f43cce7a82b2428-HKG
< content-encoding: br
```
Still no sign of the CORS header.

**Comment by yan3321** [comment]
Switching the origin to `https://localhost` and running:
```
  curl 'https://ka-f.fontawesome.com/releases/v5.15.1/css/free.min.css' \
  -X 'GET' \
  -H 'accept: */*' \
  -H 'origin: https://localhost' \
  --compressed -v
```
gets the response to include the CORS header properly, it seems:
```
< HTTP/2 200
< date: Wed, 18 Nov 2020 18:51:48 GMT
< content-type: text/css
< set-cookie: __cfduid=d03ee7c7ccdf42e3c98d19cf3db65cb7a1605725508; expires=Fri, 18-Dec-20 18:51:48 GMT; path=/; domain=.ka-f.fontawesome.com; HttpOnly; SameSite=Lax
< access-control-allow-origin: *
< access-control-allow-methods: GET
< access-control-max-age: 3000
< last-modified: Wed, 14 Oct 2020 21:18:07 GMT
< etag: W/"319d424ba89a84bbd230a3b5f7024193"
< cache-control: max-age=31556926
< access-control-allow-headers: fa-kit-token
< vary: Accept-Encoding
< x-cache: Hit from cloudfront
< via: 1.1 8de0efb362ee8f60b0c74f15a3910740.cloudfront.net (CloudFront)
< x-amz-cf-pop: HKG62-C1
< x-amz-cf-id: bEAvytyn2CyH8REZr2mc3KBiHUaILj8PKYYWVf5L96iwqxeLdYgaPA==
< age: 812969
< cf-cache-status: HIT
< cf-request-id: 067e4beafa00000be0e381e000000001
< expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=TxwDyspTYtgd7tvpvmhuQheA87OzUrehbQxbPHm6oyYRh7ZsloOAHe4jdJnFaetVY4Yc656HPoxjnstpbuOqAshakyco4%2FvqL5zGKRQSExSGmmuLpLFFTCum7CARxhimxw%3D%3D"}],"group":"cf-nel","max_age":604800}
< nel: {"report_to":"cf-nel","max_age":604800}
< server: cloudflare
< cf-ray: 5f43e28b2bab0be0-HKG
< content-encoding: br
```

**Comment by robmadole** [comment]
@yan3321 I've purged the Cloudflare cache for that resource. Can you try again?

**Comment by yan3321** [comment]
@robmadole Tried again. No change, unfortunately. Same responses.

**Comment by SBrownAU** [comment]
I've started getting this (from what seemed like) since Chrome updated during the week. Still works OK in Firefox. It happens on my localhost and my production server. I've done some digging and noticed this:

Chrome
![Selection_092](https://user-images.githubusercontent.com/4087047/99633450-d0707980-2aa3-11eb-8bd1-fd3fc1b81b2a.png)

Firefox
![Selection_090](https://user-images.githubusercontent.com/4087047/99633299-86879380-2aa3-11eb-8a98-913f09638c8b.png)

Google tells me: https://www.chromestatus.com/feature/6251880185331712

It seems like it will default to browser (in this case `strict-origin-when-cross-origin`) if header is not provided.


Chrome response headers
![Selection_089](https://user-images.githubusercontent.com/4087047/99633056-2abd0a80-2aa3-11eb-877f-62d42b13a8b4.png)


Firefox response headers
![Selection_088](https://user-images.githubusercontent.com/4087047/99633021-1aa52b00-2aa3-11eb-8ee3-04e5b3c7cae1.png)


Chrome version
```
Google Chrome | 86.0.4240.198 (Official Build) (64-bit)
-- | --
Revision | d8a506935fc2273cfbac5e5b629d74917d9119c7-refs/branch-heads/4240@{#1431}
OS | Linux
JavaScript | V8 8.6.395.25
Flash | 32.0.0.453 /home/sbrown/.config/google-chrome/PepperFlash/32.0.0.453/libpepflashplayer.so
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Safari/537.36
```

**Comment by Nick-Hope** [comment]
I have had this issue too for the last day or two, and it's intermittent. Sometimes the fonts load, sometimes not, in both Chrome & Firefox on Windows, both http and https. Here's a test page: https://bubblevision.com/fontawesome.html

**Comment by yan3321** [comment]
Issue appears to be resolved on my end.

**Comment by Nick-Hope** [comment]
Still not working for me. I switched to [self-hosting via css](https://fontawesome.com/how-to-use/on-the-web/setup/hosting-font-awesome-yourself).

**Comment by SBrownAU** [comment]
> Still not working for me. I switched to [self-hosting via css](https://fontawesome.com/how-to-use/on-the-web/setup/hosting-font-awesome-yourself).

Me neither. Your test site and console errors is exactly what I'm seeing. In lieu of any response here looks like I will also need to switch to self-hosting as my site is really not useable in many pieces of functionality without the icons.

**Comment by Nick-Hope** [comment]
My [test page](https://bubblevision.com/fontawesome.html) is now showing the characters. I didn't change anything.

**Comment by kamranayub** [comment]
Also seeing this, you can see it publicly at: https://keeptrackofmygames.com/users/kamranicus/lists/played/?sortby=Name

Seeing the issue in Edge and latest Chrome 87. It started happening in the last couple weeks. I'll probably have to switch to self-hosting.

**Comment by gevorgmakaryan** [comment]
For me the problem still exists

**Comment by gevorgmakaryan** [comment]
![image](https://user-images.githubusercontent.com/8993244/104756292-ec1e9600-575b-11eb-83be-fb8edc217021.png)

**Comment by tagliala** [comment]
hmmm

@robmadole could you please take a look? I have two different requests, one has headers, one doesn't

```
$ curl -Is https://ka-f.fontawesome.com/releases/v5.15.2/webfonts/free-fa-brands-400.woff2
HTTP/2 200 
date: Fri, 15 Jan 2021 17:15:33 GMT
content-type: font/woff2
content-length: 78476
set-cookie: __cfduid=d135c852b0eb0cfcb991efe1fa15f38d91610730933; expires=Sun, 14-Feb-21 17:15:33 GMT; path=/; domain=.ka-f.fontawesome.com; HttpOnly; SameSite=Lax
last-modified: Wed, 13 Jan 2021 18:39:04 GMT
etag: "8d08ae394e62d94154c3608ffa911f56"
cache-control: max-age=31556926
x-cache: Hit from cloudfront
via: 1.1 1a25e657bbdfbe4dde177fef28655f03.cloudfront.net (CloudFront)
x-amz-cf-pop: EWR52-C4
x-amz-cf-id: EuphymYmlp_OfsWFyoeR3JiiKtX1MtdTNax1UHF8nLWAZg59gNDS0Q==
age: 71797
cf-cache-status: HIT
accept-ranges: bytes
cf-request-id: 07a8a4a3a800003dc1c495d000000001
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=89fdhbogeDoIUbE%2F%2BgwhRskjEsXfoyZzZ%2FhcSXKD3WoD4eCb36vEhzJJk6HEuimJrB0sCxUT7R22zDiX7plc8mzaYM8uUgLLAzxUnyZOkLb6SkkfMQ%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 61213d4c3f7c3dc1-EWR

$ curl -Is https://ka-f.fontawesome.com/releases/v5.15.2/webfonts/free-fa-brands-400.woff2
HTTP/2 200 
date: Fri, 15 Jan 2021 17:15:46 GMT
content-type: font/woff2
content-length: 78476
set-cookie: __cfduid=d4f6230afbd70cfa2762a7c25f06ad21e1610730946; expires=Sun, 14-Feb-21 17:15:46 GMT; path=/; domain=.ka-f.fontawesome.com; HttpOnly; SameSite=Lax
access-control-allow-origin: *
access-control-allow-methods: GET
access-control-max-age: 3000
last-modified: Wed, 13 Jan 2021 18:39:04 GMT
etag: "8d08ae394e62d94154c3608ffa911f56"
cache-control: max-age=31556926
access-control-allow-headers: fa-kit-token
x-cache: Hit from cloudfront
via: 1.1 b9a91b9002d4fb924a73a6172edb4dc8.cloudfront.net (CloudFront)
x-amz-cf-pop: CDG3-C1
x-amz-cf-id: dR9BjijKxUuoart-_Dc5ccP-VjeUMeTHCcCccQuatavl1YHCQxCUSg==
age: 71119
cf-cache-status: HIT
accept-ranges: bytes
cf-request-id: 07a8a4d7bc0000ee602f9a3000000001
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report?s=1O%2Bg0wH%2Fr2y8ofNSzG9BbuLoWCqdKfgPV8uOCylxSlUp54yH71JyuwQnfg1FEGy3meKxXcK0zyWiLULXJUJQNSd8i3GifmL44r0M4alsXs1P1J%2BvNQ%3D%3D"}],"group":"cf-nel","max_age":604800}
nel: {"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 61213d9f9b45ee60-CDG
```

**Comment by robmadole** [comment]
Yep, looking at this now.

**Comment by robmadole** [comment]
Ok. I think I may have actually ran this one to ground. This was a nasty bit of caching (always hard to debug).

I've invalidated/purged caches and from my testing this looks fixed. Can you all check this again?

**Comment by tagliala** [comment]
> Can you all check this again?

FWIW, I've tried ~50 times and all responses had the header

Before it happened in less than 10 tries

**Comment by comdoller** [comment]
![image](https://user-images.githubusercontent.com/31316861/105010623-4f236c00-5a7f-11eb-82e6-25f465b83e9a.png)
I have the same problem. I haven't solved it yet.

**Comment by tagliala** [comment]
@comdoller please send an email at hello@fontawesome.com and provide a link to this issue

Probably we will need some extra information about to your connection to debug this issue

**Comment by Tidy1** [comment]
I'm having the same problem when I try to load the fonts from my Azure CDN endpoint.

**Comment by tagliala** [comment]
@Tidy1 could you please clarify?

Do you mean that you are hosting font awesome files on your own CDN and you have issues with CORS?

**Comment by Tidy1** [comment]
@tagliala  When I host font awesome files on my Azure CDN I get a CORS error. The rest of my CDN files (css, js, images) work without any problems. 

I am able to use the fonts only when I use them in disk space.

**Comment by tagliala** [comment]
> When I host font awesome files on my Azure CDN I get a CORS error

You should check CORS headers in your CDN

**Comment by martinrulf** [comment]
Hi shouldn't the HTTP OPTION for https://kit.fontawesome.com/272d3a8e61.js also return Access-Control-Allow-Origin header?(so as the rest of Access-Control-* headers i.e. Methods, Headers and max-Age)

If I try it with HTTP GET it does
![image](https://user-images.githubusercontent.com/37901240/125318519-32833500-e33a-11eb-959a-986f4271b9bb.png)

But for HTTP OPTION it does not
![image](https://user-images.githubusercontent.com/37901240/125318674-59416b80-e33a-11eb-856e-2ecbe86c22c4.png)

According to this
https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request
I think it should return the headers also for OPTION, elsewhere the preflight returns 404

it seems that it is the case of my scenario
![image](https://user-images.githubusercontent.com/37901240/125319007-acb3b980-e33a-11eb-8f20-1b02547385f6.png)
![image](https://user-images.githubusercontent.com/37901240/125319148-c94ff180-e33a-11eb-99cd-9f7a102af102.png)

Could you check it please?

**Comment by tagliala** [comment]
@martinrulf I've opened a new issue to track your request #18068

**Comment by Sir-Phil** [comment]
Hope this helps. Incase you're using a <script>tag. apply in index.html.

<html>
  <head>
    <script
      src="https://use.fontawesome.com/releases/v5.15.4/js/all.js"
      data-auto-a11y="true"
    ></script>
  </head>
  <body></body>
</html>

**Comment by quadratecode** [comment]
> > Hope this helps. Incase you're using a <script>tag. apply in index.html.
> > <script src="https://use.fontawesome.com/releases/v5.15.4/js/all.js" data-auto-a11y="true" ></script>
> 
> Is there a fix like this for the "all.css" file using tag?

Issue resolved for me with all.css once I upgraded from v5.15.4 to v5.15.5.

**Comment by Fyod** [comment]
May be obvious, but make sure that your kit isn't set to `www.example.com` while your users are entering from `example.com`.

**Comment by ashenthiwanka99** [comment]
> Hope this helps. Incase you're using a <script>tag. apply in index.html.
> 
> <script src="https://use.fontawesome.com/releases/v5.15.4/js/all.js" data-auto-a11y="true" ></script>

This work for  me thanks

**Comment by robmadole** [comment]
Our Kits service now supports access control headers for OPTIONS requests.

**Comment by shabscreativecontent** [comment]
> 

Hello

how can i debug this error

Access to CSS stylesheet at 'https://kit.fontawesome.com/62f846aca1.css' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

 GET https://kit.fontawesome.com/62f846aca1.css net::ERR_FAILED 404

**Comment by anjared** [comment]
> > > Hope this helps. Incase you're using a <script>tag. apply in index.html.
> > > <script src="https://use.fontawesome.com/releases/v5.15.4/js/all.js" data-auto-a11y="true" ></script>
> > 
> > 
> > Is there a fix like this for the "all.css" file using tag?
> 
> Issue resolved for me with all.css once I upgraded from v5.15.4 to v5.15.5.

I'm seeing the latest version of v5 is 5.15.4?  https://github.com/FortAwesome/Font-Awesome/releases 

We are having this CORS issue as well and are on 5.15.1.

**Comment by tagliala** [comment]
Any chance to provide a reproducible test case?

**Comment by anjared** [comment]
We just turned on the PWA of our Angular app and we are getting this error.  We only seem to be getting this for "some" users.  Right now we've only experienced it on mobile Android Chrome.

![image](https://github.com/FortAwesome/Font-Awesome/assets/108483271/64ed2920-e842-45eb-bf96-188e96574910)

We have this in our ngsw-config.json file:

```
{
  "$schema": "./node_modules/@angular/service-worker/config/schema.json",
  "index": "/index.html",
  "appData": { },
  "assetGroups": [
    {
      "name": "app",
      "installMode": "prefetch",
      "resources": {
        "files": [
          "/index.html",
......
        ],
        "urls": [
            "https://fonts.googleapis.com/css?family=Roboto:300,400,500&amp;display=swap",
            "https://fonts.googleapis.com/css?family=Roboto+Slab:300",
            "https://fonts.googleapis.com/icon?family=Material+Icons",
            "https://fonts.cdnfonts.com/css/chalkduster",
            "https://pro.fontawesome.com/releases/v5.15.1/css/all.css"
        ]
      }
    },
......
```

**Comment by tagliala** [comment]
@robmadole I can see that Pro and Free CDNs are not adding anymore

```
Access-Control-Allow-Origin: *
```

to the response. Is this expected?

**Comment by daanjipping** [comment]
Running into the same problem with the Pro version (v5.15.1) on Chrome (Version 113.0.5672.126 (Official Build) (arm64))

**Comment by anjared** [comment]
> Running into the same problem with the Pro version (v5.15.1) on Chrome (Version 113.0.5672.126 (Official Build) (arm64))

We ended up doing this as a workaround: 

"https://pro.fontawesome.com/releases/v5.15.4/css/all.css?ngsw-bypass=true"

in the urls section of the `assetGroups`.  We aren't really solving the true issue by doing this though.

**Comment by robmadole** [comment]
I just cleared the purged the Cloudflare cache for:

- https://pro.fontawesome.com/releases/v5.15.1/css/all.css
- https://pro.fontawesome.com/releases/v5.15.4/css/all.css

Make sure that any requests to these resources have the `Origin` header. That's required in order for the `Access-Control-*` headers to show up.

**Comment by malcolmocean** [comment]
I suddenly got issues from localhost today, having not had such issues a few weeks ago (didn't notice anything in between but I think it's fairly recent)

ran this
```
curl -I https://kit.fontawesome.com/65032baf49.js \
> -X 'OPTIONS' \
  -H 'authority: ka-f.fontawesome.com' \
  -H 'accept: */*' \
  -H 'access-control-request-method: GET' \
  -H 'origin: http://localhost' \
  -H 'referer: http://localhost' \
  --compressed -v
```

and got

```
*   Trying 104.18.23.52:443...
* Connected to kit.fontawesome.com (104.18.23.52) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.0 (OUT), TLS header, Certificate Status (22):
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS header, Certificate Status (22):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS header, Finished (20):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.2 (OUT), TLS header, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=Arkansas; L=Bentonville; O=Fonticons Inc; CN=*.fontawesome.com
*  start date: Nov 22 00:00:00 2022 GMT
*  expire date: Dec 23 23:59:59 2023 GMT
*  subjectAltName: host "kit.fontawesome.com" matched cert's "*.fontawesome.com"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS RSA SHA256 2020 CA1
*  SSL certificate verify ok.
* Using HTTP2, server supports multiplexing
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* Using Stream ID: 1 (easy handle 0x55ef15bffe90)
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
> OPTIONS /65032baf49.js HTTP/2
> Host: kit.fontawesome.com
> user-agent: curl/7.81.0
> accept-encoding: deflate, gzip, br, zstd
> authority: ka-f.fontawesome.com
> accept: */*
> access-control-request-method: GET
> origin: http://localhost
> referer: http://localhost
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
* TLSv1.2 (OUT), TLS header, Supplemental data (23):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
< HTTP/2 200 
HTTP/2 200 
< date: Tue, 30 May 2023 17:31:15 GMT
date: Tue, 30 May 2023 17:31:15 GMT
< content-type: text/html; charset=utf-8
content-type: text/html; charset=utf-8
< refresh: 20
refresh: 20
< set-cookie: __cfwaitingroom=ChhGQkdzYjBxMGx0MmhqcDFoQy9kNGJBPT0SjAJMeXkvZmVnSnVFUnNURGMrOE9Hd3J3ZUo2T2h1R0xzN0kzTDE3ZWVGd2ErZVl3YW1ZWlhsa056bDFaOUtMTkZlN2JPMEtuOTI4Q0Rzc0ZZS0xGUFhwV2FlVzVnYmVNR0JEcng1eTc3THVEb1B0Y0xmNUF3bUZFZmtvRG4vU2VianJ1YVgzNjBvaWt4ZXdUbDZEWDhkSXZJR0IwL1FvanBRWmlRaU1wMTg2eGM2bjk0Zk9tM2NsWGxIZjMzNHJ4ZDFtZzdNdWlkY1FUcnRPdERPQWJSemMyaENGb0YrSWo0N3cwSVZvUHdUS3FRSHhMSXRWbmIreElTMm5mQTVoZnpzWHJEeFVDZmkwUT09; Path=/; Expires=Tue, 30 May 2023 17:36:15 GMT; HttpOnly; SameSite=Lax
set-cookie: __cfwaitingroom=ChhGQkdzYjBxMGx0MmhqcDFoQy9kNGJBPT0SjAJMeXkvZmVnSnVFUnNURGMrOE9Hd3J3ZUo2T2h1R0xzN0kzTDE3ZWVGd2ErZVl3YW1ZWlhsa056bDFaOUtMTkZlN2JPMEtuOTI4Q0Rzc0ZZS0xGUFhwV2FlVzVnYmVNR0JEcng1eTc3THVEb1B0Y0xmNUF3bUZFZmtvRG4vU2VianJ1YVgzNjBvaWt4ZXdUbDZEWDhkSXZJR0IwL1FvanBRWmlRaU1wMTg2eGM2bjk0Zk9tM2NsWGxIZjMzNHJ4ZDFtZzdNdWlkY1FUcnRPdERPQWJSemMyaENGb0YrSWo0N3cwSVZvUHdUS3FRSHhMSXRWbmIreElTMm5mQTVoZnpzWHJEeFVDZmkwUT09; Path=/; Expires=Tue, 30 May 2023 17:36:15 GMT; HttpOnly; SameSite=Lax
< vary: Accept-Encoding
vary: Accept-Encoding
< server: cloudflare
server: cloudflare
< cf-ray: 7cf8b5aeea59f4aa-YVR
cf-ray: 7cf8b5aeea59f4aa-YVR
< content-encoding: gzip
content-encoding: gzip

< 
* Excess found: excess = 859 url = /65032baf49.js (zero-length body)
* Connection #0 to host kit.fontawesome.com left intact
```

**Comment by tagliala** [comment]
I'm getting 

```
access-control-allow-origin: *
```

with the example curl request

**Comment by jirkadev** [comment]
I had the same issue as @malcolmocean half an hour ago.
I also cannot sign into my account. 
After I click on the sign-in button, I am redirected to the homepage,
without being signed-in. I see error 500 in the console.

<img width="1280" alt="Screenshot 2023-05-30 at 20 00 15" src="https://github.com/FortAwesome/Font-Awesome/assets/2588479/c44d74a4-c9ee-484d-a70c-4295e5725fe9">

**Comment by cwire4** [comment]
We are also having this issue for this file starting about 30 minutes ago https://kit.fontawesome.com/aeb1285db8.js

**Comment by researcher2** [comment]
Getting the cors error also https://kit.fontawesome.com/c05cfdedaa.js, when you follow the link directly you get a Cloudflare waiting room?

**Comment by docentYT** [comment]
The same problem on my two websites. 
```js
<script src="https://use.fontawesome.com/releases/v5.15.4/js/all.js" data-auto-a11y="true" ></script>` not work for all icons.
```

Please fix it ASAP

**Comment by tagliala** [comment]
Hello, this is a know issue, please take a look at https://status.fortawesome.com/ for updates

**Comment by Ubuntufoo** [comment]
Same issue here. <script src="https://kit.fontawesome.com/0e33780c76.js" crossorigin="anonymous"></script>

Chrome browser. Heroku hosted.

**Comment by researcher2** [comment]
Resolved for me now.

**Comment by docentYT** [comment]
It has started to work on my pages too. No errors in the console.

**Comment by jirkadev** [comment]
I would like to not have this issue in the future 👍 👍 👍 
I try to self-host my FA6 premium icons.

With FA subsetter app I will pick icons that I use in my web app.
This might have the fastest web speed.
https://fontawesome.com/docs/web/dig-deeper/subsetter
